### PR TITLE
Fix JNI build after removal of libcudf sub-libraries [skip ci]

### PR DESCRIPTION
**Pull request by jlowe**
Fixing the Java bindings build after the removal of the libcudf sub-libraries in #7107.

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -518,18 +518,7 @@
...
-                                        <include>libcudf_comms.so</include>
-                                        <include>libcudf_hash.so</include>
-                                        <include>libcudf_interop.so</include>
-                                        <include>libcudf_io.so</include>
-                                        <include>libcudf_join.so</include>
-                                        <include>libcudf_merge.so</include>
-                                        <include>libcudf_partitioning.so</include>
-                                        <include>libcudf_reductions.so</include>
-                                        <include>libcudf_replace.so</include>
-                                        <include>libcudf_rolling.so</include>
```
Comment (razajafri) [inline java/pom.xml:532]: I don't know much about the background of this. Is cudf now going to be single library? can you point me to some more references on this so I can educate myself on the background of this?

Comment (razajafri) [reply java/pom.xml:532]: nm. just saw your description

**Comment by mythrocks** [comment]
I'm :+1: on this. Kicking the tyres on this now.

**Review by revans2 — APPROVED** [review]

**Comment by mythrocks** [comment]
(Github does not allow me to add this comment because this section wasn't modified in the PR. :/)

We might need to modify the Arrow paths:
```cmake
# Find static version of Arrow lib
find_library(ARROW_LIBRARY libarrow.a
  HINTS "$ENV{ARROW_ROOT}/lib"
        "$ENV{CONDA_PREFIX}/lib"
        "${CUDF_CPP_BUILD_DIR}/arrow/install/lib")
```
The build complains with: `Arrow static libs not found. Was libcudf built with ARROW_STATIC_LIB=ON?`, unless the last path is modified as follows:
```cmake
find_library(ARROW_LIBRARY libarrow.a
  HINTS "$ENV{ARROW_ROOT}/lib"
        "$ENV{CONDA_PREFIX}/lib"
        "${CUDF_CPP_BUILD_DIR}/_deps/arrow-build/release")
```
The `ARROW_INCLUDE` path might need a similar change.

**Comment by jlowe** [comment]
> We might need to modify the Arrow paths

Thanks @mythrocks, this accidentally worked in my env because I had some old build artifacts around that it found.  I updated the hints for the Arrow paths and verified with a clean build.

**Review by mythrocks — APPROVED** [review]
LGTM!

**Comment by mythrocks** [comment]
@gpucibot merge

**Pull request merged**